javatests/arcs/core/host/AllocatorTest.kt

```diff
@@ -0,0 +1,201 @@
+package arcs.core.host
+
+import arcs.core.common.Id
+import arcs.core.data.Schema
+import arcs.core.data.SchemaDescription
+import arcs.core.data.SchemaFields
+import arcs.core.data.SchemaName
+import arcs.core.sdk.Particle
+import arcs.core.storage.driver.VolatileStorageKey
+import arcs.jvm.host.ServiceLoaderHostRegistry
+import com.google.auto.service.AutoService
+import com.google.common.truth.Truth.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+class AllocatorTest {
+    /**
+     * A test recipe, two particles
+     * [WritePerson] writes a Person to a handle
+     * [ReadPerson] reads a Person from a handle
+     * [WritePerson] runs in [WritingHost]
+     * [ReadPerson] runs in [ReadingHost]
+     *
+     * Hand translated 'compilation' roughly equivalent to:
+     *
+     * schema Person { name: Text }
+     * particle ReadPerson in 'arcs.core.host.AllocatorTest.ReadPerson'
+     *   person: reads Person
+     *
+     * particle WritePerson in 'arcs.core.host.AllocatorTest.WritePerson'
+     *   person: writes Person
+     *
+     * recipe WriteAndReadPerson
+     *   recipePerson: create
+     *   WritePerson
+     *     person: writes recipePerson
+     *   ReadPerson
+     *     person: reads recipePerson
+     */
+    private val personSchema = Schema(
+        listOf(SchemaName("Person")),
+        SchemaFields(setOf("name"), emptySet()),
+        SchemaDescription()
+    )
+
+    @Target(AnnotationTarget.CLASS)
+    @Retention(AnnotationRetention.RUNTIME)
+    @TargetHost(WritingHost::class)
+    annotation class RunInWritingHost
+
+    @Target(AnnotationTarget.CLASS)
+    @Retention(AnnotationRetention.RUNTIME)
+    @TargetHost(ReadingHost::class)
+    annotation class RunInReadingHost
+
+    @AutoService(Particle::class)
+    @RunInWritingHost
+    class WritePerson : Particle
+
+    @AutoService(Particle::class)
+    @RunInReadingHost
+    class ReadPerson : Particle
+
+    open class TestingHost : AbstractArcHost() {
+        var started = mutableListOf<PlanPartition>()
+
+        override fun startArc(partition: PlanPartition) {
+            super.startArc(partition)
+            started.add(partition)
+        }
+
+        fun setup() {
+            started.clear()
+        }
+    }
+
+    @AutoService(ArcHost::class)
+    class WritingHost : TestingHost()
+
+    @AutoService(ArcHost::class)
+    class ReadingHost : TestingHost()
+
+    private lateinit var personHandleSpec: HandleSpec
```

Comment:
just for readability sake, could you, please, add a comment with the recipe (in arcs format) that you are using here?

Reply:
There's already a comment at the top of the file?

Reply:
my bad, missed it, sorry.